File: workout-service/src/main/java/com/gym/workoutservice/entity/WorkoutLog.java
package com.gym.workoutservice.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "workout_log")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class WorkoutLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "member_id", nullable = false)
    private Long memberId; // Member ID from User Service

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exercise_id", nullable = false)
    private Exercise exercise;

    @Column(name = "log_date", nullable = false)
    private LocalDate logDate;

    @Column(name = "actual_sets")
    private Integer actualSets;

    @Column(name = "actual_reps")
    private String actualReps; // Can be '8-12', '10', 'Max'

    private String notes;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: fithub-frontend/src/types/WorkoutLog.ts
import { type Exercise } from "./Exercise";

export interface WorkoutLog {
  id: number;
  memberId: number;
  exercise: Exercise; // Nested Exercise details
  logDate: string; // ISO date string
  actualSets?: number;
  actualReps?: string;
  notes?: string;
  createdAt: string;
  updatedAt: string;
}

export interface WorkoutLogRequest {
  memberId: number;
  exerciseId: number;
  logDate: string; // ISO date string
  actualSets?: number;
  actualReps?: string;
  notes?: string;
}
